pkg/apis/planetscale/v2/vitesscluster_types.go

```diff
@@ -148,6 +152,17 @@ const (
 	ImmediateVitessClusterUpdateStrategyType VitessClusterUpdateStrategyType = "Immediate"
 )
 
+type ExternalVitessClusterUpdateStrategyOptions struct {
+	// AllowResourceChanges can be used to allow certain resource updates to the DataVolumeClaimTemplate
+	// to be updated immediately without using an external tool.
+	//
+	// Supported options:
+	// - storage
+	//
+	// Default: Update no fields immediately.
+	AllowResourceChanges []string
```

Comment:
Let's use the official "enum" type for resources since we plan to match this up against the keys in a `corev1.ResourceList`. Also, all API fields need JSON tags.

```suggestion
	AllowResourceChanges []corev1.ResourceName `json:"allowResourceChanges,omitempty"`
```

Reply:
Done.